worker/mocks_test.go

```diff
@@ -113,33 +163,111 @@ func (os *mockObjectStore) Object(ctx context.Context, bucket, path string, opts
 		return api.ObjectsResponse{}, errObjectNotFound
 	}
 
-	object := os.objects[bucket][path]
+	// clone to ensure the store isn't unwillingly modified
+	var o object.Object
+	if b, err := json.Marshal(os.objects[bucket][path]); err != nil {
+		panic(err)
+	} else if err := json.Unmarshal(b, &o); err != nil {
+		panic(err)
+	}
+
 	return api.ObjectsResponse{Object: &api.Object{
-		ObjectMetadata: api.ObjectMetadata{Name: path, Size: object.TotalSize()},
-		Object:         object,
+		ObjectMetadata: api.ObjectMetadata{Name: path, Size: o.TotalSize()},
+		Object:         o,
 	}}, nil
 }
 
-func (os *mockObjectStore) DeleteHostSector(ctx context.Context, hk types.PublicKey, root types.Hash256) error {
-	return nil
-}
-func (os *mockObjectStore) DeleteObject(ctx context.Context, bucket, path string, opts api.DeleteObjectOptions) error {
-	return nil
-}
 func (os *mockObjectStore) FetchPartialSlab(ctx context.Context, key object.EncryptionKey, offset, length uint32) ([]byte, error) {
-	return nil, nil
+	os.mu.Lock()
+	defer os.mu.Unlock()
+
+	packedSlab, exists := os.partials[key.String()]
+	if !exists {
+		return nil, errSlabNotFound
+	}
+	if offset+length > uint32(len(packedSlab.data)) {
+		return nil, errors.New("offset out of bounds")
+	}
+
+	return packedSlab.data[offset : offset+length], nil
 }
+
 func (os *mockObjectStore) Slab(ctx context.Context, key object.EncryptionKey) (object.Slab, error) {
-	return object.Slab{}, nil
+	os.mu.Lock()
+	defer os.mu.Unlock()
+
+	for _, objects := range os.objects {
+		for _, object := range objects {
+			for _, slab := range object.Slabs {
+				if slab.Slab.Key.String() == key.String() {
+					return slab.Slab, nil
```

Comment:
![](https://i.pinimg.com/originals/99/d5/dc/99d5dc5603ba75faf30ce7741294e988.gif)

Reply:
It wasn't too bad imo but I added `os.forEachObject` which I could re-use 3 times.